elements/pf-popover/pf-popover.ts

```diff
@@ -0,0 +1,387 @@
+ *          Default alert icon color
+ *          {@default `#2b9af3`}
+ * @cssprop {<color>} --pf-c-popover--m-warning__title-text--Color
+ *          Default alert heading color
+ *          {@default `#795600`}
+ * @cssprop {<color>} --pf-c-popover--m-warning__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#f0ab00`}
+ * @cssprop {<color>} --pf-c-popover--m-success__title-text--Color
+ *          Default alert heading color
+ *          {@default `#1e4f18`}
+ * @cssprop {<color>} --pf-c-popover--m-success__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#3e8635`}
+ * @cssprop {<color>} --pf-c-popover--m-danger__title-text--Color
+ *          Default alert heading color
+ *          {@default `#a30000`}
+ * @cssprop {<color>} --pf-c-popover--m-danger__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#c9190b`}
+ */
+@customElement('pf-popover')
+export class PfPopover extends LitElement {
+  static readonly styles = [styles];
+
+  // todo: alertSeverityScreenReaderText
+  // todo: hideOnOutsideClick
+  // todo: withFocusTrap
+  @property({ type: String, reflect: true }) position: Placement = 'top';
+  @property({ type: String, reflect: true }) heading?: string;
+  @property({ type: String, reflect: true }) body = '';
+  @property({ type: String, reflect: true }) footer?: string;
+  @property({ type: String, reflect: true }) icon?: string;
+  @property({ type: String, reflect: true }) label?: string;
+  @property({ type: Number, reflect: true }) distance?: number = 25;
+  // todo: handle PF4s 'flip' Placement option in flip-behavior if possible
+  // https://www.patternfly.org/v4/components/popover#popover
+  @property({ type: String, attribute: 'flip-behavior', converter: StringListConverter }) flipBehavior?: Placement[] = ['top', 'bottom', 'left', 'right', 'top-start', 'top-end', 'bottom-start', 'bottom-end', 'left-start', 'left-end', 'right-start', 'right-end'];
+  @property({ type: Boolean, reflect: true, attribute: 'enable-flip' }) enableFlip?: boolean = true;
+  @property({ type: Number, reflect: true, attribute: 'heading-level' }) headingLevel?: HeadingLevel;
+  @property({ type: String, reflect: true, attribute: 'icon-set' }) iconSet?: string;
+  @property({ type: Boolean, reflect: true, attribute: 'hide-close' }) hideClose?: boolean;
+  @property({ type: String, reflect: true, attribute: 'alert-severity' }) alertSeverity?: AlertSeverity;
+  @property({ type: String, reflect: true, attribute: 'close-label' }) closeButtonLabel = 'Close popover';
+
+  @observed
+  @property({ type: String, reflect: true }) trigger?: string;
+
+  @query('#popover') private _popover?: HTMLDialogElement | null;
+  @query('#trigger') private _slottedTrigger?: HTMLElement | null;
+
+  #referenceTrigger?: HTMLElement | null = null;
+
+  #float = new FloatingDOMController(this, {
+    content: () => this.shadowRoot?.querySelector('#popover'),
```

Comment:
Do we need these query selectors in here or can we use the `_popover` variable?  Same for the arrow, but w/ a @query for the arrow element?

Reply:
Yep, I think you're right!